src/main/java/com/minecolonies/coremod/colony/ColonyManager.java

```diff
@@ -992,58 +1009,160 @@ public static void onWorldLoad(@NotNull final World world)
         }
     }
 
-    public static boolean backupColonyData()
+    /**
+     * Initiates a timer to schedule the colony backup task
+     * backupDelay of 0 disables automatic backup
+     */
+    private static void scheduleColonyBackup()
+    {
+        final int backupDelay = Configurations.administration.colonyBackupTimeMins;
+        if (backupDelay > 0)
+        {
+            backupTimerTask = new BackupColonyTimerTask();
+            backupTimer.schedule(backupTimerTask, backupDelay * MILLISECONDS_PER_MINUTE);
+        }
+    }
+
+    /**
+     * Task to be called from backupTimer, triggers main backup method
+     * by overriding run() method of base TimerTask
+     */
+
+    private static class BackupColonyTimerTask extends TimerTask
+    {
+        @Override
+        public void run()
+        {
+            Log.getLogger().info("Initiating scheduled backup");
+            backupColonyData();
+        }
+    }
+
+    /**
+     * Thread safe method to lock backups so only one backup may run at a time
+     *
+     * @return boolean indicating whether safe to initiate backup
+     */
+    private static synchronized boolean isSafeToBackup()
     {
-        if (numWorldsLoaded > 0 && saveNeeded)
+        if (backupInProgress)
         {
-            saveColonies(false);
+            Log.getLogger().warn("Cannot initiate backup - already in progress");
+            return false;
         }
+        else
+        {
+            backupInProgress = true;
+            return true;
+        }
+    }
 
-        try(FileOutputStream fos = new FileOutputStream(getBackupSaveLocation(new Date())))
+    public static boolean backupColonyData()
+    {
+        // Possibility of the timer kicking in while the command backup is being run, or vice versa
+        // Prevent by requiring a threadsafe token, reject any other backup attempts while the token
+        // is checked out - why run two backups back to back?
+        if (isSafeToBackup())
         {
-            @NotNull final File saveDir = new File(DimensionManager.getWorld(0).getSaveHandler().getWorldDirectory(), FILENAME_MINECOLONIES_PATH);
-            final ZipOutputStream zos = new ZipOutputStream(fos);
+            // Cancel any pending scheduled backup - we'll reset the timer after we're done
+            if (backupTimerTask != null)
+            {
+                backupTimerTask.cancel();
+            }
 
-            for (int i = 1; i < colonies.getTopID() + 1; i++)
+            if (numWorldsLoaded > 0 && saveNeeded)
             {
-                @NotNull final File file = new File(saveDir, String.format(FILENAME_COLONY, i));
-                if (file.exists())
+                saveColonies(false);
+            }
+            // todo Stop taking a gamble that the filename doesn't already exist, it's possible
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)